pipes/WDL/tasks/tasks_assembly.wdl

```diff
@@ -866,26 +870,47 @@ task run_discordance {
           outf.write(str(n_rgs)+'\n')
         with open('num_libraries', 'wt') as outf:
           outf.write(str(n_lbs)+'\n')
+
+        # detect empty fasta situation and manually create empty VCF
+        import os.path
+        if (os.path.getsize('~{reference_fasta}') == 0):
+          sample_name = [[x[3:] for x in h if x.startswith('SM:')][0] for h in header if h[0]=='@RG'][0]
+          with open('everything.vcf', 'wt') as outf:
+              outf.write('##fileformat=VCFv4.3')
+              outf.write('##ALT=<ID=*,Description="Represents allele(s) other than observed.">')
+              outf.write('##INFO=<ID=DP,Number=1,Type=Integer,Description="Raw read depth">')
+              outf.write('##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">')
+              outf.write('##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Number of high-quality bases">')
+              outf.write('\t'.join(('#CHROM','POS','ID','REF','ALT','QUAL','FILTER','INFO','FORMAT',sample_name))+'\n')
         CODE
 
-        # bcftools call snps while treating each RG as a separate sample
-        bcftools mpileup \
-          -G readgroups.txt -d 10000 -a "FORMAT/DP,FORMAT/AD" \
-          -q 1 -m 2 -Ou \
-          -f "~{reference_fasta}" "~{reads_aligned_bam}" \
-          | bcftools call \
-          -P 0 -m --ploidy 1 \
-          --threads $(nproc) \
-          -Ov -o everything.vcf
-
-        # mask all GT calls when less than 3 reads
-        cat everything.vcf | bcftools filter -e "FMT/DP<~{min_coverage}" -S . > filtered.vcf
-        cat filtered.vcf | bcftools filter -i "MAC>0" > "~{out_basename}.discordant.vcf"
-
-        # tally outputs
-        bcftools filter -i 'MAC=0' filtered.vcf | bcftools query -f '%POS\n' | wc -l | tee num_concordant
-        bcftools filter -i 'TYPE="snp"'  "~{out_basename}.discordant.vcf" | bcftools query -f '%POS\n' | wc -l | tee num_discordant_snps
-        bcftools filter -i 'TYPE!="snp"' "~{out_basename}.discordant.vcf" | bcftools query -f '%POS\n' | wc -l | tee num_discordant_indels
+        if [ ! -f everything.vcf ]; then
+          # bcftools call snps while treating each RG as a separate sample
+          bcftools mpileup \
+            -G readgroups.txt -d 10000 -a "FORMAT/DP,FORMAT/AD" \
```

Comment:
Do we want too parameterize the max depth here? (`-d`) and maybe also annotate with strand bias p-value? ([`FORMAT/SP`](https://samtools.github.io/bcftools/bcftools.html#mpileup))

Reply:
So for both this and the below comment -- my goal in this PR wasn't actually to improve the pre-existing script in this Task, but to just wrap it all in an `if` block and add the proper empty-file-handling code to an `else` block.

But we could parameterize `-d` I think the goal of this particular bit when we wrote it a while ago was simply to do quick and dirty identification of discordance within-sample/between-libraries, so it's not really variant calling for its own sake, but for finding potential lab problems.